crates/loader/src/assets.rs

```diff
@@ -76,18 +76,22 @@ pub(crate) fn component_dir(id: &str) -> String {
     // if the asset paths are also long. Longer term, consider an alternative approach where
     // we use an index or something for disambiguation, and/or disambiguating only if a clash is
     // detected, etc.
-    let id_sha256 = format!("{:x}", Sha256::digest(id));
+    let id_sha256 = bytes_sha256_digest_string(id.as_bytes());
     format!("{}_{}", UNSAFE_CHARACTERS.replace_all(id, "_"), id_sha256)
 }
 
+/// Return the hex-encoded SHA256 digest of the given bytes.
+pub fn bytes_sha256_digest_string(bytes: &[u8]) -> String {
+    let digest_value = Sha256::digest(bytes);
+
+    format!("{:x}", digest_value)
+}
+
 /// Return the hex-encoded SHA256 digest of the given file.
-pub fn file_sha256_digest_string(path: impl AsRef<Path>) -> std::io::Result<String> {
-    let mut file = std::fs::File::open(path)?;
-    let mut sha = sha2::Sha256::new();
-    std::io::copy(&mut file, &mut sha)?;
-    let digest_value = sha.finalize();
-    let digest_string = format!("{:x}", digest_value);
-    Ok(digest_string)
+pub async fn file_sha256_digest_string(path: impl AsRef<Path>) -> std::io::Result<String> {
+    let bytes = tokio::fs::read(&path).await?;
+
+    Ok(bytes_sha256_digest_string(&bytes))
```

Comment:
The reason for using `copy` was, I think, that the data could be streamed from the file into the hash algorithm, without needing to load a potentially large file into memory.  Unless `copy` does a full load, I'd suggest reverting to the earlier implementation even if it does result in a minor divergence of the bytes and file functions.

Reply:
You're right, I've checked it and it seems that hash algo doesn't need the whole file in memory. Reverted, thanks!